source/elements/oneMKL/source/domains/dft/descriptor.rst

```diff
@@ -417,6 +422,83 @@ type ``oneapi::mkl::dft::domain``, ``oneapi::mkl::dft::precision``,
 
 **Descriptor class member table:** :ref:`onemkl_dft_descriptor_member_table`
 
+.. _onemkl_dft_descriptor_set_workspace:
+
+set_workspace
++++++++++++++
+
+Set the workspace for when ``config_param::WORKSPACE_PLACEMENT`` is set to ``config_value::WORKSPACE_EXTERNAL``.
+
+.. rubric:: Description
+
+This function sets the workspace to use when computing DFTs for when an
+external workspace is set. 
+This function may only be called after the descriptor has been committed.
+The size of the provided workspace must be equal to or larger than the required 
+workspace size obtained by calling ``descriptor<prec, dom>::get_value(config_param::WORKSPACE_EXTERNAL_BYTES_REQUIRED, &workspaceBytes)``.
+
+A descriptor where ``WORKSPACE_EXTERNAL`` is specified is not a valid descriptor 
+for compute calls until this function has been successfully called.
+
+The type of workspace must match the compute calls for which it is used.
+That is, if the workspace is provided as a ``sycl::buffer``, the compute
+calls must also use ``sycl::buffer`` for their arguments. Likewise, a USM
+allocated workspace must only be used with USM compute calls.
+Failing to do this will result in an invalid descriptor for compute calls.
+
+If the workspace is a USM allocation, the user must not use it for other purposes
+in parallel whilst the DFT `compute_forward` or `compute_backward` are in progress.
+
+This function can be called on committed descriptors where the workspace placement
+is not ``config_value::WORKSPACE_EXTERNAL``. The provided workspace may or may not
+be used in compute calls. However, the aforementioned restrictions will still apply.
+
+.. rubric:: Syntax (buffer workspace)
+
+.. code-block:: cpp
+
+   namespace oneapi::mkl::dft {
+
+      template <oneapi::mkl::dft::precision prec, oneapi::mkl::dft::domain dom>
+      void descriptor<prec,dom>::set_workspace(sycl::buffer<scalar_type, 1> &workspaceBuf);
+   }
+
+.. rubric:: Syntax (USM workspace)
+
+.. code-block:: cpp
+
+   namespace oneapi::mkl::dft {
+
+      template <oneapi::mkl::dft::precision prec, oneapi::mkl::dft::domain dom>
+      void descriptor<prec,dom>::set_workspace(scalar_type* workspaceUSM);
+
+   }
+
+.. container:: section
+
+   .. rubric:: Input Parameters
+
+   workspaceBuf
```

Comment:
I didn't consider this, and I don't understand how a 1D sub-buffer could cause problems. However, I've added the restriction that a sub-buffer cannot be used in [cb59477](https://github.com/oneapi-src/oneAPI-spec/pull/509/commits/cb594776fba3b73c325e9233abdc1ad6ddd30e39).

The reasoning is as follows:
* The use of sub-buffers is not common.
* Closed-source oneMKL is an important backend, but does not support this feature.
* The main library supporting `sycl::buffers` in the interface is the closed-source oneMKL. Therefore codes being ported to the oneMKL interface library, that depend on the buffer interface, are likely to have been using closed-source oneMKL. Therefore, ported codes probably already support this restriction.

Reply:
Thank you!
FYI, if you want to know/understand the limitation we face with that: the workspace that we're given is conceptually divided in non-overlapping chunks attributed to separate parts of the descriptors, e.g., twiddle tables, scratch space for transposition, scratch space of children descriptors, etc. When using buffers, it is done by creating sub-buffers off the given workspace. However, one cannot create a sub-buffer from another sub-buffer, hence the restriction.

Reply:
I believe we can lift this limitation in closed source oneMKL by using ranged accessors instead of sub-buffers. If so, do we still want to keep this limitation in the spec?

Reply:
I think we want to keep this limitation for now:
* This should work with the oneMKL that can be downloaded today.
* Its easier to remove this constraint in the future instead of deciding it needs to be added back in.
* I've not seen sub-buffers used that much, so I'd be interested to know how many people this limitation affects.

Reply:
Sounds good to me as we want it to work with oneMKL as of today.